test/irb/test_history.rb

```diff
@@ -169,15 +182,17 @@ def assert_history(expected_history, initial_irb_history, input)
           f.write(initial_irb_history)
         end
 
-        io = TestInputMethodWithHistory.new
+        io = input_method.new
         io.class::HISTORY.clear
         io.load_history
         if block_given?
-          history = io.class::HISTORY.dup
+          previous_history = []
+          io.class::HISTORY.each { |line| previous_history << line }
           yield IRB.rc_file("_history")
-          io.class::HISTORY.replace(history)
+          io.class::HISTORY.clear
+          previous_history.each { |line| io.class::HISTORY << line }
         end
-        io.class::HISTORY.concat(input.split)
+        input.split.each { |line| io.class::HISTORY << line }
```

Comment:
`Readline::HISTORY` doesn't respond to `concat`.  Updated how this adds the new lines.